### DISCUSSION TITLE
Custom field - list

### DISCUSSION THREAD

**Discussion by Kani999** [netbox-community/netbox#9674] [Ideas]
Hello, 

I've run into a situation where I would like to define a custom field as a list of strings. There is a possibility only for a selection, which does not fulfill my needs because I don't know the pre-defined values for it. 
With currently defined [CF](https://docs.netbox.dev/en/stable/customization/custom-fields/) I cannot implement it. 

I would like to suggest adding the CF of type List.

**Comment by candlerb** [comment]
As a workaround for now, I'd suggest you use a text field containing a comma-separated list of values.  If the values themselves need to include comma, then either parse the line as a CSV row, or use a different separator.

**Reply by Kani999** [reply]
Yes, that's exactly what I'm doing right now. Using a separator and then parsing it in the scripts.

**Comment by jeremystretch** [comment]
You can just use the JSON type to store a list of strings.

**Reply by Kani999** [reply]
That's true, but I believe that you are not able to filter records through the WEB UI via the JSON field.

**Comment by MalfuncEddie** [comment]
I've got a similar situation where it is in a json field and would like to have a json path like filter on the column in the gui so I can filter what is shown on the gui.

**Comment by MalfuncEddie** [comment]
Some think like this?
https://github.com/netbox-community/netbox/discussions/10569#discussioncomment-3808160
